salt/addons/psp/manifests/10-podsecuritypolicy-lockdown.yaml

```diff
@@ -0,0 +1,82 @@
+# The lockdown PodSecurityPolicy is intended to be as locked
+# down as is possible. It's likely to be so restrictive as to not generally
+# be usable.
+---
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: lockdown
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default
+    seccomp.security.alpha.kubernetes.io/defaultProfileName: docker/default
+    apparmor.security.beta.kubernetes.io/allowedProfileNames: runtime/default
+    apparmor.security.beta.kubernetes.io/defaultProfileName: runtime/default
+spec:
+  # Privileged
+  privileged: false
+  # Volumes and File Systems
+  volumes:
+    # Kubernetes Pseudo Volume Types
+    - configMap
+    - secret
+    - emptyDir
+    - downwardAPI
+    - projected
+    - persistentVolumeClaim
+  allowedFlexVolumes: []
+  allowedHostPaths:
+    # Note: We don't allow hostPath volumes above, but set this to a path we
+    # control anyway as a belt+braces protection. /dev/null may be a better
+    # option, but the implications of pointing this towards a device are
+    # unclear.
+    - pathPrefix: /opt/kubernetes-hostpath-volumes
```

Comment:
> If we have to point this at a path, I would point it at one that we know is going to never be used (and is obvious should never be used). Something like /tmp/caasp-kubernetes-locked-down-hostpath

Yea - I like the notion of making it clear this is not intended to be useable. I'll make it something like `/invalid-kubernetes-hostpaths/` - given the RO root FS, and that name, it will be unusable and obviously unusable  - even if the hostPath volume type restriction is somehow bypassed.

> As for pointing it at a device, there theoretically shouldn't be an issue if pathPrefix handles path targets properly.

Yea, the special-caseness of host devices made me rethink my initial setting of `/dev/null` - I think you're agreeing with me here, that we should avoid referencing the device as it's unclear what that would end up allowing should something, somewhere, have a bug...

Reply:
(As an aside, I just took a very quick look how `pathPrefix` is implemented and I'm a little concerned. I am going to take a closer look tomorrow...)